src/metabase/api/card.clj

```diff
@@ -458,23 +472,26 @@
    result_metadata        [:maybe qr/ResultsMetadata]
    cache_ttl              [:maybe ms/PositiveInt]
    collection_preview     [:maybe :boolean]}
-  (let [card-before-update (t2/hydrate (api/write-check Card id)
-                                       [:moderation_reviews :moderator_details])]
+  (let [card-before-update      (t2/hydrate (api/write-check Card id)
+                                            [:moderation_reviews :moderator_details])
+        maybe-switched-to-model (when (and (some? dataset) (some? type))
```

Comment:
Is it not better to keep backwards and forwards compatibility and support the presence of only one param?

```clojure
(if (some? type) (= :type model) dataset)
```